From: Michael Chan <mchan@broadcom.com>
Date: Thu, 14 Feb 2013 12:13:41 +0000
Subject: [PATCH 010/108] tg3: Set initial carrier state to off.
Patch-mainline: v3.9-rc1
Git-commit: 402e1398a398009d82731ec433220e172be4c7af
References: bsc#908458 FATE#317507


Before the device is opened, the carrier state should be off.  It
will not race with the link interrupt if we set it before calling
register_netdev().

Signed-off-by: Michael Chan <mchan@broadcom.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Acked-by: Cho, Yu-Chen <acho@suse.com>
---
 drivers/net/ethernet/broadcom/tg3.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/net/ethernet/broadcom/tg3.c
+++ b/drivers/net/ethernet/broadcom/tg3.c
@@ -16953,6 +16953,8 @@ static int tg3_init_one(struct pci_dev *
 
 	tg3_timer_init(tp);
 
+	tg3_carrier_off(tp);
+
 	err = register_netdev(dev);
 	if (err) {
 		dev_err(&pdev->dev, "Cannot register net device, aborting\n");
